### PR TITLE
ref(js): Add props to passed breadcrumbs component

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/components/breadcrumbs.tsx [hunk 1]
```diff
@@ -102,8 +103,19 @@ const BreadcrumbList = styled('div')`
   padding: ${space(1)} 0;
 `;
 
-const BreadcrumbLink = styled(({preserveGlobalSelection, ...props}) =>
-  preserveGlobalSelection ? <GlobalSelectionLink {...props} /> : <Link {...props} />
+type BreadcrumbListProps = {
+  children?: React.ReactNode;
+  preserveGlobalSelection?: boolean;
+  to: React.ComponentProps<typeof Link>['to'];
```
Comment (priscilawebdev) [inline static/app/components/breadcrumbs.tsx:109]:
```suggestion
  to: React.ComponentProps<typeof Link>['to'];
  children?: React.ReactNode;
  preserveGlobalSelection?: boolean;
```

File: static/app/components/breadcrumbs.tsx [hunk 2]
```diff
@@ -102,8 +103,19 @@ const BreadcrumbList = styled('div')`
   padding: ${space(1)} 0;
 `;
 
-const BreadcrumbLink = styled(({preserveGlobalSelection, ...props}) =>
-  preserveGlobalSelection ? <GlobalSelectionLink {...props} /> : <Link {...props} />
+type BreadcrumbListProps = {
```
Comment (priscilawebdev) [inline static/app/components/breadcrumbs.tsx:106]: should we name this type "BreadcrumbLinkProps"?

Comment (EvanPurkhiser) [reply static/app/components/breadcrumbs.tsx:106]: yeah oops

File: static/app/components/breadcrumbs.tsx [hunk 3]
```diff
@@ -102,8 +103,19 @@ const BreadcrumbList = styled('div')`
   padding: ${space(1)} 0;
 `;
 
-const BreadcrumbLink = styled(({preserveGlobalSelection, ...props}) =>
-  preserveGlobalSelection ? <GlobalSelectionLink {...props} /> : <Link {...props} />
+type BreadcrumbLinkProps = {
+  children?: React.ReactNode;
```
Comment (priscilawebdev) [inline static/app/components/breadcrumbs.tsx:107]: can we move the optional props to the end of this object?

**Review by priscilawebdev — APPROVED** [review]
left one suggestion but it looks good to me

File: static/app/components/breadcrumbs.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply static/app/components/breadcrumbs.tsx:107]: sure, will follow up

**Pull request merged**